services/director/src/simcore_service_director/registry_proxy.py

```diff
@@ -78,49 +79,44 @@ async def _basic_auth_registry_request(
     )
 
     session = get_client_session(app)
-    try:
-        async with session.request(
-            method.lower(), url, auth=auth, **session_kwargs
-        ) as response:
-            if response.status == HTTPStatus.UNAUTHORIZED:
-                logger.debug("Registry unauthorized request: %s", await response.text())
-                # basic mode failed, test with other auth mode
-                resp_data, resp_headers = await _auth_registry_request(
-                    app_settings,
-                    url,
-                    method,
-                    response.headers,
-                    session,
-                    **session_kwargs,
-                )
 
-            elif response.status == HTTPStatus.NOT_FOUND:
-                raise ServiceNotAvailableError(service_name=path)
+    response = await session.request(
+        method.lower(), f"{url}", auth=auth, **session_kwargs
+    )
 
-            elif response.status > 399:
-                logger.exception(
-                    "Unknown error while accessing registry: %s", str(response)
-                )
-                raise RegistryConnectionError(msg=str(response))
+    if response.status_code == httpx.codes.UNAUTHORIZED:
+        _logger.debug("Registry unauthorized request: %s", response.text)
+        # basic mode failed, test with other auth mode
+        resp_data, resp_headers = await _auth_registry_request(
+            app_settings,
+            url,
+            method,
+            response.headers,
+            session,
+            **session_kwargs,
+        )
 
-            else:
-                # registry that does not need an auth
-                resp_data = await response.json(content_type=None)
-                resp_headers = response.headers
-
-            return (resp_data, resp_headers)
-    except client_exceptions.ClientError as exc:
-        logger.exception("Unknown error while accessing registry")
-        msg = f"Unknown error while accessing registry: {exc!s}"
-        raise DirectorRuntimeError(msg=msg) from exc
+    elif response.status_code == httpx.codes.NOT_FOUND:
```

Comment:
To ensure consistency, I strongly recommend using `fastapi.status` instead of `httpx.codes`. While they ultimately both resolve to integers, relying on `fastapi.status` helps avoid unnecessary variations across the codebase. In fact, I even adapted `fastapi.status` for use in `aiohttp` services under `serviceslib.aiohttp.status` and developed utilities around these concepts. You can find the implementation here: [status_codes_utils.py](https://github.com/itisfoundation/osparc-simcore/blob/018075362f84489ce917699dfb4c0961d385791e/packages/service-library/src/servicelib/status_codes_utils.py).

Reply:
it's a bit weird to use fastapi status when using a different client, but ok will do.